File: Python/pset6/mario/more/mario.py
import cs50

while True:
    height = cs50.get_int("Height: ")
    if height > 0 and height < 9:
        break

left_spaces = height - 1
hashes = 1

for i in range(height):

    for b in range(left_spaces):
        print(" ", end="")

    for c in range(hashes):
        print("#", end="")

    print("  ", end="")

    for d in range(hashes):
        print("#", end="")

    print("")

    left_spaces -= 1
    hashes += 1

File: C/Problem sets/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int get_valid_input(void);

int main(void)
{
    
    int input = get_valid_input();
    int left_spaces = input - 1;
    int hashes = 1;
    //for gap i < 2 is fixed
    
    //print pyramid
    for (int height = 0; height < input; height++)
    {
        //print left spaces on current row
        for (int row = 0; row < left_spaces; row++)
        {
            printf(" ");
        }
        
        //print left hashes on current row
        for (int row = 0; row < hashes; row++)
        {
            printf("#");
        }
        
        printf("  ");
        
        //print right hashes on current row
        for (int row = 0; row < hashes; row++)
        {
            printf("#");
        }
        
        printf("\n");
    
        //calc new values
        left_spaces--;
        hashes++;
        
    }
}

//Get valid input from the user
int i;
int get_valid_input()
{
    do
    {
        i = get_int("Height: ");
    }
    while (i < 1 || i > 8);
    return i;
}